src/expression.d

```diff
@@ -5255,6 +5257,66 @@ public:
                 continue;
             }
 
+            if (auto v = s.isVarDeclaration())
+            {
+                if ((!v.type || !v.type.deco) && v._scope)
+                    v.semantic(v._scope);
+
+                if (!v.type)
+                {
+                    error("forward reference of %s %s", v.kind(), v.toChars());
```

Comment:
I know it can be hard to trigger this and the following two error messages, but can a test suite entry be done for them?